выполнено с помощью монитора

AntonKutl opened the pull request.

AntonKutl committed:
выполнено с помощью монитора

stvort commented on hw-14/src/main/java/WaitNotifyClass.java:
```diff
@@ -0,0 +1,93 @@
+        twoThread.start();
+    }
+
+    public void printOne() {
```
`printOne` и `printTwo` же одинаковые практически? Думаю можно обойтись одним методом

stvort commented on hw-14/src/main/java/WaitNotifyClass.java:
```diff
@@ -0,0 +1,93 @@
+                    Thread.sleep(50);
+
+                    str.append(i+" ");
+                    System.out.print(str);
```
При таком выводе не понятно чередуются цифры или нет. Лучше тогда в столбик выводить:
```
Поток1: 1
Поток2: 1
...
```

AntonKutl committed:
объединены методы printeOne и printeTwo в метод printe

stvort commented on hw-14/src/main/java/WaitNotifyClass.java:
```diff
@@ -0,0 +1,56 @@
+
+public class WaitNotifyClass {
+    private final Object monitor = new Object();
+    private volatile String flagTread = "one";
```
Зачем нужен `String`, как флаг? Достаточно `boolean`

stvort commented on hw-14/src/main/java/WaitNotifyClass.java:
```diff
@@ -0,0 +1,56 @@
+
+    public void print(String flag) {
+        synchronized (monitor) {
+            StringBuffer str=new StringBuffer("Поток "+ flag+":");
```
Почему не `StringBuilder`? Доступ же синхронизирован

stvort commented on hw-14/src/main/java/WaitNotifyClass.java:
```diff
@@ -0,0 +1,56 @@
+            StringBuffer str=new StringBuffer("Поток "+ flag+":");
+            try {
+
+                for (int i = 1; i < 11; i++) {
```
Можно сделать с одним циклом (с 1 до 19)

stvort commented on hw-14/src/main/java/WaitNotifyClass.java:
```diff
@@ -0,0 +1,56 @@
+                        monitor.wait();
+                    }
+
+                    str.append(i+" ");
```
Зачем конкатенировать строку если используется `StringBuffer`

stvort commented on hw-14/src/main/java/WaitNotifyClass.java:
```diff
@@ -0,0 +1,56 @@
+                    str.append(i+" ");
+                    System.out.println(str);
+                    if (flag.equals("one"))flagTread = "two";
+                    else flagTread = "one";
```
С помощью тернарного оператора можно красивее записать в одну строку

AntonKutl committed:
исправленны некоторые замечания

AntonKutl committed:
добавлен обратный счет

stvort commented on hw-14/src/main/java/WaitNotifyClass.java:
```diff
@@ -0,0 +1,41 @@
+
+public class WaitNotifyClass {
+    private final Object monitor = new Object();
+    private volatile String nameTread = "one";// блокировка потока по имени, с boolean чередование не получиться
```
Ну можно номер сделать) Строка уж как-то слишком "тяжело"

stvort commented on hw-14/src/main/java/WaitNotifyClass.java:
```diff
@@ -0,0 +1,41 @@
+
+public class WaitNotifyClass {
```
Класс лучше положить в пакет

stvort commented on hw-14/src/main/java/WaitNotifyClass.java:
```diff
@@ -0,0 +1,41 @@
+                int count=0;
+
+                for (int i = 1; i < 20; i++) {
+                    Thread.sleep(350);
```
Магические строки и числа лучше сделать константами

AntonKutl committed:
Изменения:
1) изменен тип переменной numberTread;
2) файл перемещен в пакет;
3) добавленная константа
